launchdarkly-android-client-sdk/src/main/java/com/launchdarkly/sdk/android/PollingDataSource.java

```diff
@@ -63,15 +69,16 @@ final class PollingDataSource implements DataSource {
 
     @Override
     public void start(final Callback<Boolean> resultCallback) {
-
-        if (initialDelayMillis > 0) {
-            // if there is an initial delay, we will immediately report the successful start of the data source
+        if (maxNumberOfPolls == 0) {
+            // If there are no polls to be made, we will immediately report the successful start of the data source.  This
+            // may seem strange, but one can think of this data source as behaving like a no-op in this configuration.
             resultCallback.onSuccess(true);
+            return;
         }
```

Comment:
For reviewers: The PR I opened previously to add "polling interval across restarts" used `initialDelayMillis > 0` to short circuit initialized to true.  I realized that was incorrect after examining isInitialized behavior.  This bug existed only in 5.5.0 which was released two weeks ago.

When `maxNumberOfPolls == 0`, isInitialized should be true because the flags are "as up to date as they can be in the current configuration", which is how isInitialized behaves currently.  For example: when offline mode is set, isInitialized returns true.

Reply:
TODO: add fix documentation.

Reply:
Added commit override to PR description.  Done.